File: kotlin-architect/assets/ArchitectureExample/profile-feature/src/main/kotlin/example/feature/statemachine/ProfileEvents.kt
package example.feature.statemachine

import example.domain.Profile
import example.domain.ProfileLoadFailure
import example.feature.ProfileInput
import java.util.UUID

internal sealed interface ProfileEvent

internal data class ProfileInputWasReceived(val input: ProfileInput) : ProfileEvent

internal data object ProfileLoadWasRequested : ProfileEvent

internal data object ProfileRetryWasRequested : ProfileEvent

internal data class ProfileLoadDidSucceed(
    val requestId: UUID,
    val profile: Profile,
) : ProfileEvent

internal data class ProfileLoadDidFail(
    val requestId: UUID,
    val reason: ProfileLoadFailure,
) : ProfileEvent

File: swift-architect/assets/ArchitectureExample/Sources/ProfileFeature/StateMachine/ProfileEvents.swift
import DomainModel
import Foundation
import StateMachineCore

public enum ProfileEvent: Equatable, Sendable {}

struct ProfileInputWasReceived: Equatable, Event {
  typealias SuperEvent = ProfileEvent
  let input: ProfileInput
}

struct ProfileLoadingWasRequested: Equatable, Event {
  typealias SuperEvent = ProfileEvent
}

struct ProfileRetryWasRequested: Equatable, Event {
  typealias SuperEvent = ProfileEvent
}

struct ProfileLoadingDidSucceed: Equatable, Event {
  typealias SuperEvent = ProfileEvent
  let requestID: UUID
  let profile: Profile
}

struct ProfileLoadingDidFail: Equatable, Event {
  typealias SuperEvent = ProfileEvent
  let requestID: UUID
  let failure: ProfileLoadFailure
}

struct ProfileOutcomeWasConsumed: Equatable, Event {
  typealias SuperEvent = ProfileEvent
  let deliveryID: UUID
}
